web-console/src/views/datasource-view/datasource-view.tsx

```diff
@@ -397,43 +421,43 @@ ORDER BY 1`;
 
         let unused: string[] = [];
         if (showUnused) {
-          const unusedResp = await Api.instance.get(
+          const unusedResp = await Api.instance.get<string[]>(
             '/druid/coordinator/v1/metadata/datasources?includeUnused',
           );
-          unused = unusedResp.data.filter((d: string) => !seen[d]);
+          unused = unusedResp.data.filter(d => !seen[d]);
         }
 
-        const rulesResp = await Api.instance.get('/druid/coordinator/v1/rules');
+        const rulesResp = await Api.instance.get<Record<string, Rule[]>>(
+          '/druid/coordinator/v1/rules',
+        );
         const rules = rulesResp.data;
 
-        const compactionConfigsResp = await Api.instance.get(
-          '/druid/coordinator/v1/config/compaction',
-        );
+        const compactionConfigsResp = await Api.instance.get<{
+          compactionConfigs: CompactionConfig[];
+        }>('/druid/coordinator/v1/config/compaction');
         const compactionConfigs = lookupBy(
           compactionConfigsResp.data.compactionConfigs || [],
-          (c: CompactionConfig) => c.dataSource,
+          c => c.dataSource,
         );
 
-        const compactionStatusesResp = await Api.instance.get(
+        const compactionStatusesResp = await Api.instance.get<{ latestStatus: CompactionStatus[] }>(
           '/druid/coordinator/v1/compaction/status',
         );
         const compactionStatuses = lookupBy(
           compactionStatusesResp.data.latestStatus || [],
-          (c: CompactionStatus) => c.dataSource,
+          c => c.dataSource,
         );
 
-        const allDatasources = (datasources as any).concat(
-          unused.map(d => ({ datasource: d, unused: true })),
-        );
-        allDatasources.forEach((ds: any) => {
-          ds.rules = rules[ds.datasource] || [];
-          ds.compactionConfig = compactionConfigs[ds.datasource];
-          ds.compactionStatus = compactionStatuses[ds.datasource];
-        });
-
         return {
-          datasources: allDatasources,
-          defaultRules: rules['_default'],
+          datasources: datasources.concat(unused.map(makeUnusedDatasource)).map(ds => {
```

Comment:
Does this make it so that now the `_default` datasource shows up under the list of datasources?

Reply:
no, `_default` will not be in `datasources` or `unused`, it is only a (special) key in `rules` accessed below